ai_edge_torch/generative/examples/stable_diffusion/tokenizer.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2024 The AI Edge Torch Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import functools
+import json
+import os
+from typing import List, Tuple
+import unicodedata
+
+import regex as re
+
+
+def create_bytes_table() -> dict:
+  table = {}
+  special_count = 0
+  for byte in range(256):
+    category = unicodedata.category(chr(byte))
+    if category[0] not in ['C', 'Z']:  # ith character is NOT control char or space
+      table[byte] = chr(byte)
+    else:  # ith character IS control char or space
+      table[byte] = chr(special_count + 256)
+      special_count += 1
+  return table
+
+
+def pairwise(seq):
+  a = iter(seq)
+  b = iter(seq)
+  next(b)
+  return zip(a, b)
+
+
+class Tokenizer:
```

Comment:
is this tokenizer also compatible with MP/BYOM's tokenizer?

Reply:
There is a counterpart CLIP tokenizer in mediapipe.